dlp/api/Snippets/DeidentifyWithReplacement.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright 2023 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+// [START dlp_deidentify_replace]
+
+using System;
+using System.Collections.Generic;
+using Google.Api.Gax.ResourceNames;
+using Google.Cloud.Dlp.V2;
+
+public class DeidentifyWithReplacement
+{
+    public static DeidentifyContentResponse Deidentify(
+        string projectId,
+        string text,
+        string replaceText = null,
+        IEnumerable<InfoType> infoTypes = null)
+    {
+        // Instantiate the client.
+        var dlp = DlpServiceClient.Create();
+
+        // Construct the inspect config.
+        var inspectConfig = new InspectConfig
+        {
+            InfoTypes = { infoTypes ?? new InfoType[] { new InfoType { Name = "EMAIL_ADDRESS" } } }
+        };
+
+        // Construct the replace value config.
+        var replaceConfig = new ReplaceValueConfig
+        {
+            NewValue = new Value { StringValue = replaceText ?? "[email-address]" }
+        };
+
+        // Construct the deidentify config using replace value config.
+        var deidentifyConfig = new DeidentifyConfig
+        {
+            InfoTypeTransformations = new InfoTypeTransformations
+            {
+                Transformations =
+                {
+                    new InfoTypeTransformations.Types.InfoTypeTransformation
+                    {
+                        InfoTypes = { infoTypes ?? new InfoType[] { new InfoType { Name = "EMAIL_ADDRESS" } } },
```

Comment:
Can we do this upfront (`infoTypes ??= ...`) instead of repeating this logic twice? That would make it easier to change the default value in the future and avoid mistakes where we update one location but not the other.

Reply:
Moved infotypes declaration to the top.